File: back/src/main/java/com/back/domain/member/member/service/AuthTokenService.java
package com.back.domain.member.member.service;

import com.back.domain.member.member.entity.Member;
import com.back.standard.util.Ut;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

import java.util.Map;
import java.util.UUID;

@Service
public class AuthTokenService {
    @Value("${custom.accessToken.expirationSeconds}")
    private int expireSeconds;

    @Value("${custom.jwt.secretKey}")
    private String secret;

    public String genAccessToken(Member member) {
        // JWT엔 내부 PK(Long)가 아니라 외부 공개 식별자(uuid)를 싣는다 — 순번 노출 방지.
        String id = member.getUuid().toString();
        String email = member.getEmail();
        String role = member.getRole();

        return Ut.jwt.toString(
                secret,
                expireSeconds,
                Map.of("id", id, "email", email, "role", role)
        );
    }

    public Map<String, Object> payload(String accessToken) {
        Map<String, Object> parsedPayload = Ut.jwt.payload(secret, accessToken);

        if (parsedPayload == null) return null;

        UUID uuid = UUID.fromString((String) parsedPayload.get("id"));
        String email = (String) parsedPayload.get("email");
        String role = (String) parsedPayload.get("role");

        return Map.of("id", uuid, "email", email, "role", role);
    }
}

File: back/src/main/kotlin/com/back/domain/member/member/service/AuthTokenService.kt
package com.back.domain.member.member.service

import com.back.domain.member.member.entity.Member
import com.back.standard.util.Ut
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import java.util.UUID

@Service
class AuthTokenService {
    @Value("\${custom.accessToken.expirationSeconds}")
    private var expireSeconds: Int = 0

    @Value("\${custom.jwt.secretKey}")
    private lateinit var secret: String

    fun genAccessToken(member: Member): String {
        // JWT엔 내부 PK(Long)가 아니라 외부 공개 식별자(uuid)를 싣는다 — 순번 노출 방지.
        val id = member.uuid.toString()
        val email = member.email
        val role = member.role

        return Ut.jwt.toString(
            secret,
            expireSeconds,
            mapOf("id" to id, "email" to email, "role" to role)
        )
    }

    fun payload(accessToken: String): Map<String, Any?>? {
        val parsedPayload = Ut.jwt.payload(secret, accessToken) ?: return null

        val uuid = UUID.fromString(parsedPayload["id"] as String)
        val email = parsedPayload["email"]
        val role = parsedPayload["role"]

        return mapOf("id" to uuid, "email" to email, "role" to role)
    }
}
